plane/src/database/backend.rs

```diff
@@ -585,3 +577,25 @@ impl BackendRow {
         self.as_of - self.last_status_time
     }
 }
+
+/// Update the backend_state table, without updating the backend table.
+pub async fn emit_state_change(
+    txn: &mut PgConnection,
+    backend: &BackendName,
+    new_state: &BackendState,
+) -> sqlx::Result<()> {
+    sqlx::query!(
+        r#"
+        insert into backend_state (backend_id, state)
+        values ($1, $2)
+        "#,
+        backend.to_string(),
+        serde_json::to_value(&new_state).expect("BackendState should always be JSON-serializable."),
+    )
+    .execute(&mut *txn)
+    .await?;
+
+    emit_with_key(txn, &backend.to_string(), new_state).await?;
```

Comment:
note: this is what forced the change from using a generic sqlx `Executor` to passing a `&mut PgConnection`. If we have an `Executor`, once we pass it to `execute(...)` we have moved it, so we can't use it for `emit_with_key`. Changing everything to references fixes this.